webpack/assets/javascripts/react_app/redux/API/APISelectors.js

```diff
@@ -10,8 +11,11 @@ export const selectAPIStatus = (state, key) =>
 export const selectAPIPayload = (state, key) =>
   selectAPIByKey(state, key).payload || {};
 
-export const selectAPIResponse = (state, key) =>
-  selectAPIByKey(state, key).response || {};
+export const selectAPIResponse = (state, key, asMutable = false) => {
+  const { response } = selectAPIByKey(state, key);
+  if (!response) return {};
+  return asMutable ? Immutable.asMutable(response, { deep: true }) : response;
+};
```

Comment:
My main concern here is that selectors get called many times and the asMutable operation, especially with using `deep`, is a very expensive operation, I would prefer dropping usage of seamless-immutable in the API reducer if the consumer passed the asMutable flag to the API action.

Reply:
> I would prefer dropping usage of seamless-immutable in the API reducer if the consumer passed the asMutable flag 

But then the data won't be stored immutably which can lead to redundant re-rendering while comparing states. there is a good reason to keep the state immutable.
`useSelector()` only forces a re-render if the selector result appears to be different than the last result, plus, a cached result may be returned by the hook without re-running the selector if it's the same function reference as on a previous render of the component.

on the other hand, [this asMutable](https://github.com/theforeman/foreman_ansible/pull/427/files#diff-e062f8871bd0c866e3b26d17ad37adc2d0eea5cc577ca0eebf7b1f0b68dda28dR9) will be re-run over and over on every re-render

Reply:
Thanks for sharing the use case, though I really like passing the options here,
I want to investigate if it's more efficient to set the `asMutable` in the hook itself and not on the selector, or if it's better to use it in the component e.g: `const res = response.asMutable()` only when needed after getting the results.